frontend/src/components/settings/SettingInput.tsx

```diff
@@ -153,6 +177,42 @@ const SettingInput: React.FC<RenderSettingInputProps> = ({
             {...rest}
           />
         );
+      } else if (setting.label === "target_block_id") {
```

Comment:
For both "fallback_block" and "target_block_id", the setting could have any name. So we need to check if the setting is select and that the entity is "Block".